probe-rs/src/bin/probe-rs/main.rs

```diff
@@ -173,6 +178,37 @@ fn default_logfile_location() -> Result<PathBuf> {
     Ok(log_path)
 }
 
+/// Prune all old log files in the `directory`.
+fn prune_logs(directory: &Path) -> Result<(), anyhow::Error> {
+    // Get the path and elapsed creation time of all files in the log directory that have the '.log'
+    // suffix.
+    let mut log_files = fs::read_dir(directory)?
+        .filter_map(|entry| {
+            if let Ok(entry) = entry {
+                let path = entry.path();
+                if path.extension() == Some(OsStr::new("log")) {
+                    let metadata = fs::metadata(&path).ok()?;
+                    let last_modified = metadata.created().ok()?.elapsed().ok()?.as_secs();
+                    Some((path, last_modified))
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        })
+        .collect_vec();
+
+    // Order all files by the elapsed creation time with smallest first.
+    log_files.sort_unstable_by_key(|(_, b)| *b);
+
+    // Iterate all files except for the first `MAX_LOG_FILES` and delete them.
```

Comment:
Doesn't this keep the 20 oldest files? The sort above sorts the file so that the smallest creation time comes first. It's not completely clear what smallest means here, but if it's e.g. the elapsed seconds since the EPOCH, then the first files would be the oldest ones.

Reply:
Good point, fixed!